### PR TITLE
Change Entity Namespace

### PR DESCRIPTION
**Pull request by mostafakhudair**
Change Entity class namespace
Refactor Entity casts

- [x] Component(s) with PHPdocs
- [x] User guide updated
- [x] Conforms to style guide

### CODE REVIEW

**Comment by mostafakhudair** [comment]
@paulbalandan i'm done where, would you like to review it?

**Comment by samsonasik** [comment]
What is the benefit of new namespace? next we need to move "Model" classes to Model namespace? The current usage is easier to access (no need deep namespace extends) for mostly db based apps.

**Comment by mostafakhudair** [comment]
The benefit is not about the namespace, it's about organize system directories/sub-directories

**Review by paulbalandan — APPROVED** [review]

**Comment by paulbalandan** [comment]
I'm done here. I would like @MGatner to take  final look.

File: system/Entity.php [hunk 1]
```diff
@@ -11,612 +11,22 @@
...
-	 */
-	public function cast(bool $cast = null)
-	{
-		if (null === $cast)
-		{
-			return $this->_cast;
-		}
-		$this->_cast = $cast;
-		return $this;
+		parent::__construct($data);
```
Comment (MGatner) [inline system/Entity.php:30]: What is the point of this?

Comment (mostafakhudair) [reply system/Entity.php]: Nothing,, removed :D

**Review by MGatner — APPROVED** [review]
Looks good to me! Any final requests or concerns?

**Comment by mostafakhudair** [comment]
should i add Services::entity() in a separated PR? or push it in this PR?

**Comment by MGatner** [comment]
Let's hold off on a service for now. I'd like some resolution to our interface discussion before we start adding more services for classes that don't use interfaces.

**Review by MGatner — APPROVED** [review]

**Pull request merged**